lib/vrf.c

```diff
@@ -994,25 +994,50 @@ const char *vrf_get_default_name(void)
 	return vrf_default_name;
 }
 
-int vrf_bind(vrf_id_t vrf_id, int fd, const char *name)
+int vrf_bind(vrf_id_t vrf_id, int fd, const char *ifname)
 {
 	int ret = 0;
 	struct interface *ifp;
+	struct vrf *vrf;
 
-	if (fd < 0 || name == NULL)
+	if (fd < 0)
 		return fd;
-	/* the device should exist
-	 * otherwise we should return
-	 * case ifname = vrf in netns mode => return
-	 */
-	ifp = if_lookup_by_name(name, vrf_id);
-	if (!ifp)
+
+	if (vrf_id == VRF_UNKNOWN)
 		return fd;
+
+	/* can't bind to a VRF that doesn't exist */
+	vrf = vrf_lookup_by_id(vrf_id);
+	if (!vrf_is_enabled(vrf))
+		return -1;
```

Comment:
if you reached that code, dont you forget to close previous fd ?

Reply:
This function is not supposed to close the socket. It should be done in the calling function (`vrf_socket`).

Reply:
Yep, I don't see any resources we need to release here

Reply:
ok